README.md

```diff
@@ -24,209 +41,46 @@ from discoverx import DX
-```
-
-Note: You need to regularly [vacuum](https://docs.delta.io/latest/delta-utility.html#remove-files-no-longer-referenced-by-a-delta-table) all your delta tables to remove all traces of your deleted rows. 
-
-### Select
-
-Select all columns classified with specified classes from multiple tables
-
-```
-dx.select_by_classes(from_tables="*.*.*", by_classes=["dx_iso_date", "dx_email"], min_score=None)
-```
-
-You can apply further transformations to build your summary tables. 
-Eg. Count the occurrence of each IP address per day across multiple tables and columns
-
-```
-df = (dx.select_by_classes(from_tables="*.*.*", by_classes=["dx_iso_date", "dx_ip_v4"])
-    .groupby(["table_catalog", "table_schema", "table_name", "classified_columns.dx_iso_date.column", "classified_columns.dx_iso_date.value", "classified_columns.dx_ip_v4.column"])
-    .agg(func.count("classified_columns.dx_ip_v4.value").alias("count"))
-)
-```
-
+## Available functionality
 
-## Configuration
+The available `dx` functions are
 
-### Scan parameters
+* `from_tables("<catalog>.<schema>.<table>")` selects tables based on the specified pattern (use `*` as a wildcard). Returns a [DataExplorer](#dataexplorer-object) object. 
+* `intro` gives an introduction to the library
+* `scan` scans the lakehouse with regex expressions defined by the rules and to power the semantic classification. [Documentation](docs/Semantic_classification.md)
+* `display_rules` shows the rules available for semantic classification
+* `search` searches the lakehouse content for by leveraging the semantic classes identified with scan (eg. email, ip address, etc.). [Documentaiton](docs/Search.md)
+* `select_by_class` selects data from the lakehouse content by semantic class. [Documentation](docs/Select_by_class.md)
+* `delete_by_class` deletes from the lakehouse by semantic class. [Documentation](docs/Delete_by_class.md)
 
-You can define 
+### DataExplorer Object
 
-```
-dx.scan(
-    from_tables="*.*.*", # Table pattern in form of <catalog>.<schema>.<table> ('*' is a wildcard)
-    rules="*",           # Rule filter ('*' is a wildcard)
-    sample_size=10000,   # Number of rows to sample, use None for a full table scan
-    what_if=False        # If `True` it prints the SQL that would be executed
-)
-```
+The functions available from the `DataExplorer` Object are:
 
-### Custom rules
+* `having_columns` restricts the selection to tables that have the specified columns
+* `with_concurrency` defines how many queries are executed concurrently (10 by defailt)
+* `apply_sql` applies a SQL template to all tables and returns a [DataExplorerActions](#dataexploreractions-object) object
+* `unpivot_string_columns` returns a melted (unpivoted) dataframe with all string columns from the selected tables and returns a [DataExplorerActions](#dataexploreractions-object) object
 
-You can provide your custom scanning rules based on regex expressions.
+### DataExplorerActions Object
```

Comment:
Same as with DataExplorer Object ...

Reply:
Nesting this section would get it too complex, check out how it looks now after I moved the DataExplorer docs.

Reply:
I think it's fine to have it here but I would describe it more on a high level and not refer to `DataExplorerActions`. Rather describe HOW you can use `explain` and `execute`, i.e. that you can chain these and give a very short example. It's not important at this point which object they belong to as this makes it a bit too complex I think.

Reply:
Ok, got it! 
Then let's merge this as it is, and I will open a new one with this change.